spec/services/payment_providers/adyen/customers/create_service_spec.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe PaymentProviders::Adyen::Customers::CreateService, type: :service do
+  let(:create_service) { described_class.new(customer:, payment_provider_id:, params:, async:) }
+
+  let(:customer) { create(:customer) }
+  let(:adyen_provider) { create(:adyen_provider, organization: customer.organization) }
+  let(:payment_provider_id) { adyen_provider.id }
+  let(:params) { {provider_customer_id: "id", sync_with_provider: true} }
+  let(:async) { true }
+
+  describe ".call" do
+    it "creates a payment_provider_customer" do
+      result = create_service.call
+
+      expect(result).to be_success
+      expect(result.provider_customer).to be_present
+      expect(result.provider_customer.provider_customer_id).to eq("id")
+    end
+
+    context "when no provider customer id and should create on service" do
+      let(:params) do
+        {provider_customer_id: nil, sync_with_provider: true}
+      end
+
+      it "enqueues a job to create the customer on the provider" do
+        expect { create_service.call }.to have_enqueued_job(PaymentProviderCustomers::AdyenCreateJob)
+      end
+    end
+
+    context "when removing the provider customer id and should create on service" do
```

Comment:
Actually, the `should create on service` refers to the presence of the `sync_with_provider` attribute. And since we are removing the `provider_customer_id`, we want to make sure that the provider is not reached in that case.